### PR TITLE
enh(ci): adding sonar analysis for all maintenance branches

### PR DESCRIPTION
**Pull request by zguennoune02**
## Description

Adding sonar analysis for centreon-frontend code.

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [x] 20.04.x
- [x] 20.10.x
- [x] 21.04.x
- [x] 21.10.x (master)

<h2> How this pull request can be tested ? </h2>

Launch a build from centreon-frontend In Jenkins "Centreon Frontend" Job.

### CODE REVIEW

**Review by Thebarda — APPROVED** [review]

File: Jenkinsfile [hunk 1]
```diff
@@ -43,40 +43,39 @@ def checkoutCentreonBuild(buildBranch) {
...
       env.VERSION = "${source.VERSION}"
       env.RELEASE = "${source.RELEASE}"
-      stash includes: '**', name: 'centreon-frontend-uicontext-centreon-build'
-    }
+      sh "./centreon-build/jobs/frontend/${serie}/frontend-sources.sh"
+      stash includes: '**', name: 'centreonui-centreon-build'
+      stash includes: '**', name: 'uicontext-centreon-build'
   }
+  if ((currentBuild.result ?: 'SUCCESS') != 'SUCCESS') {
+      error('Quality gate stage failure');
```
Comment (sc979) [inline Jenkinsfile:70]: erro('Sonar analysis...

File: sonar-project.properties [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# project
+sonar.projectKey={PROJECT_TITLE}
+sonar.projectName={PROJECT_NAME}
+sonar.projectVersion={PROJECT_VERSION}
```
Comment (sc979) [inline sonar-project.properties:4]: should the files containing the test be anaysed too ?
If not, you need to specify the test files location and source location

File: Jenkinsfile (hunk 1 above)
Comment (zguennoune02) [reply Jenkinsfile:70]: Thanks

File: sonar-project.properties (hunk 2 above)
Comment (zguennoune02) [reply sonar-project.properties:4]: Test files should be excluded. I did the change with sonar.exclusions.

**Review by sc979 — APPROVED** [review]

**Pull request merged**